### PR TITLE
docs: add container executor docs

### PR DESCRIPTION
**Pull request by povilasv**
## Pull request description 

Adding docs for https://github.com/kubeshop/testkube/pull/2293 pull request.

Ref https://github.com/kubeshop/testkube/issues/2133

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [x] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

**Review by exu — APPROVED** [review]

File: docs/docs/4-test-types/container-executor.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+---
+sidebar_position: 11
+sidebar_label: Container Executor
+---
+# What is a Container Executor?
+
+TestKube container executor allows you to run your own container images for executing tests. TestKube orchestrates the Tests using the container image as Kubernetes Jobs.
```
Comment (jfermi) [inline docs/docs/4-test-types/container-executor.md:7]:
```suggestion
The TestKube container executor allows you to run your own container images for executing tests. TestKube orchestrates the Tests using the container image as Kubernetes Jobs.
```

File: docs/docs/4-test-types/container-executor.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+The Test execution fails if the container exits with an error and succeeds when the container command successfully executes.
+
+In order to use container executor, create a new executor with `executor_type: container` and your custom type. For example:
```
Comment (jfermi) [inline docs/docs/4-test-types/container-executor.md:11]:
```suggestion
In order to use the container executor, create a new executor with `executor_type: container` and your custom type. For example:
```

File: docs/docs/4-test-types/container-executor.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  executionRequest:
+    args:
+    - https://testkube.kubeshop.io/
+    envs:
+      TESTKUBE_ENV: example
+```
+
+Executing this test will run a Kubernetes Job with a `curlimages/curl:7.85.0` image, `secret-name` image pull secret, `curl` command, and `https://testkube.kubeshop.io/` argument.
+
+You can provide image, args, command, and image pull secrets in HTTP Request, Test Spec, and Executor spec. The container executor merges all the data using the following order:
```
Comment (jfermi) [inline docs/docs/4-test-types/container-executor.md:48]:
```suggestion
You can provide image, args, command, and image pull secrets in the HTTP Request, Test Spec, and Executor spec. The container executor merges all the data using the following order:
```

**Review by jfermi — COMMENTED** [review]
Some editing suggestions.

**Pull request merged**